src/app/admin/modules/program-management/admin-programs/components/admin-program-list/admin-program-list.component.ts

```diff
@@ -19,32 +21,43 @@ export class AdminProgramListComponent implements OnInit {
   departments: Department[] = [];
   /** The total number of programs. */
   isLoadingResults = true;
-
+  isSchoolAdmin = this.authService.isSchoolAdmin;
   constructor(
     private readonly route: ActivatedRoute,
     private readonly router: Router,
     private readonly programService: ProgramService,
+    private readonly departmentService: DepartmentService,
+    @Inject(AUTH_SERVICE) private readonly authService: AuthService,
   ) { }
 
   ngOnInit() {
-    this.programService.getPrograms({offset: 0, limit: 100}).pipe(
-      map(data => {
-        this.isLoadingResults = false;
-        this.programs = data.results;
-        return this.programs;
-      }),
-      catchError((err) => {
-        this.isLoadingResults = false;
-        return observableOf([]);
-      })).subscribe(
-      data => {
-        this.programs = data;
+    if (this.isSchoolAdmin) {
+      this.departmentService.getTopDepartments().pipe(
+        catchError((err) => {
+          this.isLoadingResults = false;
+          return observableOf([]);
+        })
+      ).subscribe(data => {
+          this.isLoadingResults = false;
+          this.departments = data;
       });
-  }
+    } else {
+      this.loadProgramsBelongToDepartment(this.authService.department);
+      this.isLoadingResults = false;
+    }
+}
 
   navigateToRelatedEvents(row: Program) {
     this.router.navigate(['../../events'], { queryParams: {program_id: row.id}, relativeTo: this.route});
   }
 
+  loadProgramsBelongToDepartment(department: number) {
+    const extraParams = new Map();
+    extraParams.set('department', department);
+    this.programService.getPrograms({offset: 0, limit: 200, extraParams}).subscribe(
```

Comment:
Change all occurrences about `limit=200`, use `limit=-1` for no pagination instead.